statistics/handle/lru_cache.go

```diff
@@ -65,6 +69,13 @@ type lruMapElement struct {
 	tblMemUsage *statistics.TableMemoryUsage
 }
 
+func (l *lruMapElement) copy() *lruMapElement {
+	return &lruMapElement{
+		tbl:         l.tbl,
+		tblMemUsage: l.tblMemUsage,
+	}
+}
```

Comment:
Using copy here is necessary, otherwise it will cause data race like following:

```text
[2022-05-07T04:44:38.619Z] WARNING: DATA RACE
[2022-05-07T04:44:38.619Z] Write at 0x00c00a7b61a0 by goroutine 75:
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*statsInnerCache).put()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/lru_cache.go:117 +0xd1
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*statsInnerCache).Put()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/lru_cache.go:109 +0xc4
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.statsCache.update()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/statscache.go:91 +0x1e1
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*Handle).Update()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/handle.go:303 +0x32b
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/domain.(*Domain).loadStatsWorker()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/domain/domain.go:1390 +0x725
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/domain.(*Domain).loadStatsWorker-fm()
[2022-05-07T04:44:38.619Z]       <autogenerated>:1 +0x39
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/wait_group_wrapper.go:33 +0x73
[2022-05-07T04:44:38.619Z] 
[2022-05-07T04:44:38.619Z] Previous read at 0x00c00a7b61a0 by goroutine 156:
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*statsInnerCache).Get()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/lru_cache.go:95 +0x10f
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*Handle).TableStatsFromStorage()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/handle.go:841 +0x167
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*Handle).Update()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/handle.go:285 +0x8cd
[2022-05-07T04:44:38.619Z]   github.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON()
[2022-05-07T04:44:38.619Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/statistics/handle/dump.go:228 +0x3a8
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/executor.(*LoadStatsInfo).Update()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/executor/load_stats.go:90 +0x25b
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).handleLoadStats()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:1756 +0xd7
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).handleQuerySpecial()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:2073 +0x34d
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).handleStmt()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:2045 +0x588
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).handleQuery()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:1875 +0x94f
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).dispatch()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:1370 +0x11f1
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*clientConn).Run()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/conn.go:1120 +0x2f3
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*Server).onConn()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/server.go:556 +0x1b84
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*Server).startNetworkListener.func2()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/server.go:453 +0x47
[2022-05-07T04:44:38.620Z] 
[2022-05-07T04:44:38.620Z] Goroutine 75 (running) created at:
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/wait_group_wrapper.go:31 +0xdc
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/domain.(*Domain).UpdateTableStatsLoop()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/domain/domain.go:1319 +0x27c
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/domain.(*Domain).LoadAndUpdateStatsLoop()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/domain/domain.go:1299 +0x6b
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/session.BootstrapSession()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/session.go:2908 +0xac4
[2022-05-07T04:44:38.620Z]   main.createStoreAndDomain()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/tidb-server/main.go:298 +0x113
[2022-05-07T04:44:38.620Z]   main.main()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/tidb-server/main.go:204 +0x5e9
[2022-05-07T04:44:38.620Z] 
[2022-05-07T04:44:38.620Z] Goroutine 156 (running) created at:
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*Server).startNetworkListener()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/server.go:453 +0x6f3
[2022-05-07T04:44:38.620Z]   github.com/pingcap/tidb/server.(*Server).Run.func1()
[2022-05-07T04:44:38.620Z]       /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/server/server.go:368 +0x65
```